### PR TITLE
[Feature Branch] Add Compass Rose and Compass HUD to Imagery View

### PR DESCRIPTION
**Pull request by davetsay**
## Overview
Addresses #3658 
merging into feature branch `imagery-enhancements`

## Testing Instructions 
see #3658 

## Author Checklist
| | |
| --- | :---: |
| Changes address original issue? | Y |
| Unit tests included and/or updated with changes? | N/A |
| Command line build passes? | Y |
| Changes have been smoke-tested? | Y |
| Testing instructions included? | Y |

### CODE REVIEW

**Comment by davetsay** [comment]
@akhenry , @jvigliotta , @charlesh88 , this is only merging into feature branch but it should be easier to review this chunk independently.

File: src/plugins/imagery/components/Compass/Compass.vue [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        // degrees from north heading
+        sunHeading() {
+            return this.image['Sun Orientation'];
+        },
+        // degrees from rover heading
+        cameraPan() {
+            return this.image['Camera Pan'];
+        },
+        cameraTilt() {
+            return this.image['Camera Tilt'];
```
Comment (deeptailor) [inline src/plugins/imagery/components/Compass/Compass.vue:104]: Is there a reason why we are not using snake case for keys?

File: src/plugins/imagery/components/ImageryViewLayout.vue [hunk 2]
```diff
@@ -199,13 +237,20 @@ export default {
             }
 
             return result;
+        },
+        shouldDisplayCompass() {
+            return this.focusedImage !== undefined
+                && this.focusedImageNaturalAspectRatio !== undefined
+                && this.imageContainerWidth !== undefined
+                && this.imageContainerHeight !== undefined;
```
Comment (deeptailor) [inline src/plugins/imagery/components/ImageryViewLayout.vue:245]: We could add these variables into an array and do: VARIABLES.all(variable => variable !== undefined), so we are not repeating logic.

**Review by deeptailor — CHANGES_REQUESTED** [review]
Looks great, just a couple of comments

**Pull request merged**